odoo/models.py

```diff
@@ -4814,14 +4828,11 @@ def with_env(self, env):
         """
         return self._browse(env, self._ids, self._prefetch_ids)
 
-    def sudo(self, user=SUPERUSER_ID):
-        """ sudo([user=SUPERUSER])
-
-        Returns a new version of this recordset attached to the provided
-        user.
+    def sudo(self, mode=True):
```

Comment:
maybe `sudo(self, mode=True, user=None)` and also check case for `if user:` -> show warning and return `self.with_user(user)`